grammar.js

```diff
@@ -29,25 +23,37 @@ module.exports = grammar({
     $.comment,
   ],
 
+  precedences: $ => [
+    // TODO: more-specific operator precedence
+    [$.binary_operation, $.assignment],
+    // TODO: figure out why foo.bar = 42 currently adds an "(ERROR)" node
+    [$.assignment, $.property],
+  ],
```

Comment:
This is a much nicer way to represent relative precedence compared to the sort of "z-index inflation" of keeping track of which numbers should be higher than others: https://github.com/tree-sitter/tree-sitter/pull/939